src/context/cart-context.js

```diff
@@ -1,17 +1,20 @@
-import { createContext, useContext, useState } from "react";
+import { createContext, useContext, useState, useEffect } from "react";
 import { useReducer } from "react";
 import { cartReducer } from "../reducers";
-
+import { getCartDataHandler } from "../utils";
 const defaultCartContext = {
 	type: "",
 	cartItemsCount: 0,
 	itemsInCart: [],
+	cartData: [],
 };
 
 const CartContext = createContext({ defaultCartContext });
 
 const CartProvider = ({ children }) => {
 	const [cartState, cartDispatch] = useReducer(cartReducer, defaultCartContext);
+	const token = localStorage.getItem("token");
```

Comment:
You can create auth context and store the token in auth context and get it from there instead of getting it from localstorage everytime